packages/protocol/contracts/dapis/SamplePriceDataFeed.sol

```diff
@@ -2,15 +2,33 @@
 pragma solidity 0.8.4;
 
 import "./RrpDapiServer.sol";
+import "./Api3Adminship.sol";
 
 // A sample price data feed implemented as a dAPI client
-contract SamplePriceDataFeed {
+contract SamplePriceDataFeed is Api3Adminship {
   RrpDapiServer public rrpDapiServer;
+
   bytes16 public latestDapiId;
   int256 public latestAnswer;
 
-  constructor(address _rrpDapiServer) {
+  uint256 public cooldownTime = 1 days; // TODO: is this the right value?
```

Comment:
@bbenligiray

Reply:
I don't know, coming up with the specs is the hardest part of this and depends on what you envision a dAPI operation would look like. The main concern is the admins attacking, so I guess erring on the side of a longer cooldown period is better (1 week?).